torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -690,6 +690,10 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
   void performNocolorSplit(at::Device device);
 
+  void registerUserBuffers(at::Device device);
+
+  void deregisterUserBuffers(at::Device device);
+
```

Comment:
Do you mind adding descriptions to these two public methods?

Reply:
Re the name of the APIs:
after reading the API impl, it seems that we are registering all segments of the context mem pool. So I wonder if "UserBuffers" in the API names are from NCCL perspective rather than from PyTorch perspective?

Reply:
Re the argument to the API:
any preference between passing a `device` or a `memPool`?
If accepting a `device`, it implies that the API will get the mem pool from current context?